packages/arb-avm-cpp/data_storage/src/arbcore.cpp

```diff
@@ -2471,10 +2472,12 @@ bool ArbCore::logsCursorConfirmReceived(size_t cursor_index) {
         return false;
     }
 
-    ReadWriteTransaction tx(data_storage);
-    auto status = logsCursorSaveCurrentTotalCount(
-        tx, cursor_index, logs_cursors[cursor_index].pending_total_count);
-    tx.commit();
+    if (logs_cursors[cursor_index].deleted_data.empty()) {
```

Comment:
Great catch, thanks

Reply:
I've removed those checks. They should be replaced by the new state check. Since we can afford to miss confirmations, if there's a reorg inbetween getting and confirming, we simply skip the confirmation and let go process the reorg before recording the next confirmation.